test/TestUtils.py

```diff
@@ -526,7 +526,45 @@ def test_nested_items() -> None:
         ("list-item", "apple", "fruits"),
         ("list-item", "orange", "fruits"),
     ]
-    assert list(utils.nested_items(data)) == items
+    with pytest.deprecated_call(
+        match=r"Call to deprecated function ansiblelint\.utils\.nested_items.*"
+    ):
+        assert list(utils.nested_items(data)) == items
+
+
+def test_nested_items_path() -> None:
+    """Verify correct function of nested_items_path()."""
+    data = {
+        "foo": "text",
+        "bar": {"some": "text2"},
+        "fruits": ["apple", "orange"],
+        "answer": [{"forty-two": ["life", "universe", "everything"]}],
+    }
+
+    items = [
+        ("foo", "text", []),
+        ("bar", {"some": "text2"}, []),
+        ("some", "text2", ["bar"]),
+        ("fruits", ["apple", "orange"], []),
+        (0, "apple", ["fruits"]),
+        (1, "orange", ["fruits"]),
+        ("answer", [{"forty-two": ["life", "universe", "everything"]}], []),
+        (0, {"forty-two": ["life", "universe", "everything"]}, ["answer"]),
+        ("forty-two", ["life", "universe", "everything"], ["answer", 0]),
+        (0, "life", ["answer", 0, "forty-two"]),
+        (1, "universe", ["answer", 0, "forty-two"]),
+        (2, "everything", ["answer", 0, "forty-two"]),
+    ]
+    assert list(utils.nested_items_path(data)) == items
+
+
+def test_nested_items_path_raises_typeerror() -> None:
```

Comment:
It'd look like this:
```suggestion
@pytest.mark.parametrize(
    'invalid_data_input',
    (
        "string",
        42,
        1.234,
        None,
        ("tuple",),
        {"set"},
    )
)
def test_nested_items_path_raises_typeerror(invalid_data_input) -> None:
```

Reply:
done